.circleci/config.yml

```diff
@@ -1,68 +1,370 @@
 version: 2.1
 
 commands:
+  checkout-shallow:
+    description: "Do a shallow checkout of the repository"
+    steps:
+      - run:
+          name: Install git and openssh
+          command: apt-get update && apt-get -qq -y install openssh-client git
+      - run:
+          name: Store the keyid for the git repo
+          command: mkdir -p $HOME/.ssh && ssh-keyscan $(echo ${CIRCLE_REPOSITORY_URL} | sed 's,.*@\([^:]*\):.*,\1,') > $HOME/.ssh/known_hosts
```

Comment:
Any reason we are doing this instead of telling git/ssh to just don't care about the key? Or, alternatively, why not do an https checkout?

Reply:
This is mostly done to keep the git checkout very close to what circle is doing by default (SSH checkout)